// @discord-tickets/settings@2.4.0/build/server/chunks/11-C02GIQv0.js
import { _ as __variableDynamicImportRuntimeHelper } from './dynamic-import-helper-uMTE3ehW.js';
import { i as importJSON } from './i18n-Cdh88uwz.js';

async function load({ parent }) {
  const { locale } = await parent();
  return {
    translations: importJSON(await __variableDynamicImportRuntimeHelper(/* @__PURE__ */ Object.assign({ "../../../lib/locales/en-GB/_common.json": () => import('./_common-DF670JSI.js') }), `../../../lib/locales/${locale}/_common.json`, 7))
  };
}

var _page = /*#__PURE__*/Object.freeze({
  __proto__: null,
  load: load
});

const index = 11;
let component_cache;
const component = async () => component_cache ??= (await import('./_page.svelte-7jdXqexP.js')).default;
const universal_id = "src/routes/(default)/[guild]/+page.js";
const imports = ["_app/immutable/nodes/11.DmWfrhZc.js","_app/immutable/chunks/preload-helper.C1FmrZbK.js","_app/immutable/chunks/I18n.DS7_lcac.js","_app/immutable/chunks/disclose-version.Bg9kRutz.js","_app/immutable/chunks/runtime.D2_qJMqH.js","_app/immutable/chunks/index.BFqC5wTN.js","_app/immutable/chunks/render.DYyZBxIN.js","_app/immutable/chunks/utils.Dghph3y9.js","_app/immutable/chunks/svelte-head.BxbvRlmQ.js","_app/immutable/chunks/template.-1lJmaN6.js","_app/immutable/chunks/if.DQDEx0g4.js","_app/immutable/chunks/attributes.HKi8Fu_N.js","_app/immutable/chunks/store.B-TJ19st.js","_app/immutable/chunks/utils.ZaDWqmk0.js","_app/immutable/chunks/stores.DEmN4aYk.js","_app/immutable/chunks/entry.DD3pBHu9.js","_app/immutable/chunks/index-client.BK-sKD8p.js","_app/immutable/chunks/paths.C8eWa2kM.js","_app/immutable/chunks/control.CYgJF_JY.js"];
const stylesheets = [];
const fonts = [];

export { component, fonts, imports, index, stylesheets, _page as universal, universal_id };
