// @discord-tickets/settings@2.4.0/build/server/chunks/15-D7p7QfuU.js
async function load({ fetch }) {
  const fetchOptions = { credentials: "include" };
  return {
    guilds: await (await fetch(`/api/admin/guilds`, fetchOptions)).json()
  };
}

var _page = /*#__PURE__*/Object.freeze({
  __proto__: null,
  load: load
});

const index = 15;
let component_cache;
const component = async () => component_cache ??= (await import('./_page.svelte-Cso8qepN.js')).default;
const universal_id = "src/routes/settings/+page.js";
const imports = ["_app/immutable/nodes/15.DBuAWVJ5.js","_app/immutable/chunks/disclose-version.Bg9kRutz.js","_app/immutable/chunks/runtime.D2_qJMqH.js","_app/immutable/chunks/index.BFqC5wTN.js","_app/immutable/chunks/render.DYyZBxIN.js","_app/immutable/chunks/utils.Dghph3y9.js","_app/immutable/chunks/svelte-head.BxbvRlmQ.js","_app/immutable/chunks/template.-1lJmaN6.js","_app/immutable/chunks/if.DQDEx0g4.js","_app/immutable/chunks/each.DhYb_c6Z.js","_app/immutable/chunks/attributes.HKi8Fu_N.js","_app/immutable/chunks/paths.C8eWa2kM.js"];
const stylesheets = [];
const fonts = [];

export { component, fonts, imports, index, stylesheets, _page as universal, universal_id };
