// @discord-tickets/settings@2.4.0/build/server/chunks/17-BARUUtYf.js
import { e as error } from './index2-DzcLzHBX.js';

async function load({ fetch, params }) {
  const response = await fetch(`/api/admin/guilds/${params.guild}/categories`);
  const isJSON = response.headers.get("Content-Type")?.includes("json");
  const body = isJSON ? await response.json() : await response.text();
  if (!response.ok) {
    error(response.status, isJSON ? JSON.stringify(body) : body);
  } else {
    return { categories: body };
  }
}

var _page = /*#__PURE__*/Object.freeze({
  __proto__: null,
  load: load
});

const index = 17;
let component_cache;
const component = async () => component_cache ??= (await import('./_page.svelte-C9wSRZ4J.js')).default;
const universal_id = "src/routes/settings/[guild]/categories/+page.js";
const imports = ["_app/immutable/nodes/17.Boj5nXEK.js","_app/immutable/chunks/index.g5YcAAdQ.js","_app/immutable/chunks/control.CYgJF_JY.js","_app/immutable/chunks/disclose-version.Bg9kRutz.js","_app/immutable/chunks/runtime.D2_qJMqH.js","_app/immutable/chunks/index.BFqC5wTN.js","_app/immutable/chunks/render.DYyZBxIN.js","_app/immutable/chunks/utils.Dghph3y9.js","_app/immutable/chunks/svelte-head.BxbvRlmQ.js","_app/immutable/chunks/template.-1lJmaN6.js","_app/immutable/chunks/if.DQDEx0g4.js","_app/immutable/chunks/each.DhYb_c6Z.js","_app/immutable/chunks/attributes.HKi8Fu_N.js","_app/immutable/chunks/index.B101_2Qz.js","_app/immutable/chunks/_commonjsHelpers.Cpj98o6Y.js"];
const stylesheets = [];
const fonts = [];

export { component, fonts, imports, index, stylesheets, _page as universal, universal_id };
