// @discord-tickets/settings@2.4.0/build/server/chunks/18-nvEMqBvZ.js
import { e as error } from './index2-DzcLzHBX.js';

async function load({ fetch, params }) {
  const fetchOptions = { credentials: "include" };
  let body;
  if (params.category === "new") {
    body = {
      channelName: "",
      claiming: false,
      description: "",
      discordCategory: "new",
      enableFeedback: false,
      emoji: "",
      image: "",
      memberLimit: 1,
      name: "",
      openingMessage: "",
      pingRoles: [],
      questions: [],
      ratelimit: null,
      requiredRoles: [],
      requireTopic: false,
      staffRoles: [],
      totalLimit: 50
    };
  } else {
    const response = await fetch(
      `/api/admin/guilds/${params.guild}/categories/${params.category}`,
      fetchOptions
    );
    const isJSON = response.headers.get("Content-Type")?.includes("json");
    body = isJSON ? await response.json() : await response.text();
    if (!response.ok) {
      error(response.status, isJSON ? JSON.stringify(body) : body);
    }
  }
  let url = `/api/admin/guilds/${params.guild}/categories`;
  if (params.category !== "new") url += `/${params.category}`;
  return {
    url,
    category: body,
    channels: await (await fetch(`/api/admin/guilds/${params.guild}/data?query=channels.cache`, fetchOptions)).json(),
    roles: await (await fetch(`/api/admin/guilds/${params.guild}/data?query=roles.cache`, fetchOptions)).json(),
    settings: await (await fetch(`/api/admin/guilds/${params.guild}/settings`, fetchOptions)).json()
  };
}

var _page = /*#__PURE__*/Object.freeze({
  __proto__: null,
  load: load
});

const index = 18;
let component_cache;
const component = async () => component_cache ??= (await import('./_page.svelte-sMYpvzJc.js')).default;
const universal_id = "src/routes/settings/[guild]/categories/[category]/+page.js";
const imports = ["_app/immutable/nodes/18.CD87Cf1l.js","_app/immutable/chunks/index.g5YcAAdQ.js","_app/immutable/chunks/control.CYgJF_JY.js","_app/immutable/chunks/preload-helper.C1FmrZbK.js","_app/immutable/chunks/disclose-version.Bg9kRutz.js","_app/immutable/chunks/runtime.D2_qJMqH.js","_app/immutable/chunks/index.BFqC5wTN.js","_app/immutable/chunks/render.DYyZBxIN.js","_app/immutable/chunks/utils.Dghph3y9.js","_app/immutable/chunks/svelte-head.BxbvRlmQ.js","_app/immutable/chunks/template.-1lJmaN6.js","_app/immutable/chunks/if.DQDEx0g4.js","_app/immutable/chunks/each.DhYb_c6Z.js","_app/immutable/chunks/html.BiDgUgRj.js","_app/immutable/chunks/attributes.HKi8Fu_N.js","_app/immutable/chunks/Required.0zfnBhzh.js","_app/immutable/chunks/legacy.CtaTdtmd.js","_app/immutable/chunks/select.DchCu2yw.js","_app/immutable/chunks/proxy.CdfeBWYI.js","_app/immutable/chunks/event-modifiers.BDD76Zze.js","_app/immutable/chunks/index-client.BK-sKD8p.js","_app/immutable/chunks/legacy-client.bwmqDaLy.js","_app/immutable/chunks/attributes.CaPm3nup.js","_app/immutable/chunks/index.D3WHGDsz.js","_app/immutable/chunks/_commonjsHelpers.Cpj98o6Y.js","_app/immutable/chunks/index.B101_2Qz.js","_app/immutable/chunks/data.CSunYKlX.js","_app/immutable/chunks/class.CZ_Lwgno.js","_app/immutable/chunks/props.DioveIhQ.js","_app/immutable/chunks/store.B-TJ19st.js","_app/immutable/chunks/utils.ZaDWqmk0.js","_app/immutable/chunks/stores.DEmN4aYk.js","_app/immutable/chunks/entry.DD3pBHu9.js","_app/immutable/chunks/paths.C8eWa2kM.js","_app/immutable/chunks/Modals.CkctU4O8.js","_app/immutable/chunks/snippet.DPPw8-wN.js","_app/immutable/chunks/svelte-component.BdD4hKdB.js","_app/immutable/chunks/state.svelte.ciRFsZAd.js","_app/immutable/chunks/ErrorBox.CbGQpU3f.js"];
const stylesheets = [];
const fonts = [];

export { component, fonts, imports, index, stylesheets, _page as universal, universal_id };
