// @discord-tickets/settings@2.4.0/build/server/chunks/21-C4I6cN91.js
import { e as error } from './index2-DzcLzHBX.js';

async function load({ fetch, params }) {
  const fetchOptions = { credentials: "include" };
  const response = await fetch(`/api/admin/guilds/${params.guild}/categories`, fetchOptions);
  const isJSON = response.headers.get("Content-Type")?.includes("json");
  const body = isJSON ? await response.json() : await response.text();
  if (!response.ok) {
    error(response.status, isJSON ? JSON.stringify(body) : body);
  } else {
    return {
      categories: body,
      channels: await (await fetch(`/api/admin/guilds/${params.guild}/data?query=channels.cache`, fetchOptions)).json()
    };
  }
}

var _page = /*#__PURE__*/Object.freeze({
  __proto__: null,
  load: load
});

const index = 21;
let component_cache;
const component = async () => component_cache ??= (await import('./_page.svelte-ChvjGxVM.js')).default;
const universal_id = "src/routes/settings/[guild]/panels/+page.js";
const imports = ["_app/immutable/nodes/21.FYFcXd8i.js","_app/immutable/chunks/index.g5YcAAdQ.js","_app/immutable/chunks/control.CYgJF_JY.js","_app/immutable/chunks/disclose-version.Bg9kRutz.js","_app/immutable/chunks/runtime.D2_qJMqH.js","_app/immutable/chunks/index.BFqC5wTN.js","_app/immutable/chunks/render.DYyZBxIN.js","_app/immutable/chunks/utils.Dghph3y9.js","_app/immutable/chunks/svelte-head.BxbvRlmQ.js","_app/immutable/chunks/template.-1lJmaN6.js","_app/immutable/chunks/if.DQDEx0g4.js","_app/immutable/chunks/each.DhYb_c6Z.js","_app/immutable/chunks/attributes.HKi8Fu_N.js","_app/immutable/chunks/Required.0zfnBhzh.js","_app/immutable/chunks/legacy.CtaTdtmd.js","_app/immutable/chunks/select.DchCu2yw.js","_app/immutable/chunks/proxy.CdfeBWYI.js","_app/immutable/chunks/event-modifiers.BDD76Zze.js","_app/immutable/chunks/store.B-TJ19st.js","_app/immutable/chunks/utils.ZaDWqmk0.js","_app/immutable/chunks/legacy-client.bwmqDaLy.js","_app/immutable/chunks/stores.DEmN4aYk.js","_app/immutable/chunks/entry.DD3pBHu9.js","_app/immutable/chunks/index-client.BK-sKD8p.js","_app/immutable/chunks/paths.C8eWa2kM.js","_app/immutable/chunks/index.B101_2Qz.js","_app/immutable/chunks/_commonjsHelpers.Cpj98o6Y.js","_app/immutable/chunks/ErrorBox.CbGQpU3f.js","_app/immutable/chunks/class.CZ_Lwgno.js","_app/immutable/chunks/props.DioveIhQ.js","_app/immutable/chunks/data.CSunYKlX.js","_app/immutable/chunks/html.BiDgUgRj.js","_app/immutable/chunks/index.DF_JIHC5.js","_app/immutable/chunks/transitions.BIDcdJtW.js","_app/immutable/chunks/lifecycle.J282GXZ8.js"];
const stylesheets = ["_app/immutable/assets/21.SOp4b2K_.css"];
const fonts = [];

export { component, fonts, imports, index, stylesheets, _page as universal, universal_id };
