// @discord-tickets/settings@2.4.0/build/server/chunks/7-C3FSOYGW.js
import { e as error } from './index2-DzcLzHBX.js';

async function load({ fetch, params }) {
  const response = await fetch(`/api/admin/guilds/${params.guild}`);
  const isJSON = response.headers.get("Content-Type")?.includes("json");
  const body = isJSON ? await response.json() : await response.text();
  if (!response.ok) {
    error(response.status, isJSON ? JSON.stringify(body) : body);
  } else {
    return { guild: body };
  }
}

var _layout = /*#__PURE__*/Object.freeze({
  __proto__: null,
  load: load
});

const index = 7;
let component_cache;
const component = async () => component_cache ??= (await import('./_layout.svelte-CK7TMepn.js')).default;
const universal_id = "src/routes/settings/[guild]/+layout.js";
const imports = ["_app/immutable/nodes/7.CSvd8feB.js","_app/immutable/chunks/index.g5YcAAdQ.js","_app/immutable/chunks/control.CYgJF_JY.js","_app/immutable/chunks/disclose-version.Bg9kRutz.js","_app/immutable/chunks/runtime.D2_qJMqH.js","_app/immutable/chunks/index.BFqC5wTN.js","_app/immutable/chunks/render.DYyZBxIN.js","_app/immutable/chunks/utils.Dghph3y9.js","_app/immutable/chunks/svelte-head.BxbvRlmQ.js","_app/immutable/chunks/template.-1lJmaN6.js","_app/immutable/chunks/snippet.DPPw8-wN.js"];
const stylesheets = [];
const fonts = [];

export { component, fonts, imports, index, stylesheets, _layout as universal, universal_id };
