// @discord-tickets/settings@2.4.0/build/server/chunks/_layout.svelte-6uGhJs3V.js
import { p as push, a1 as head, $ as ensure_array_like, a2 as once, s as setContext, a as pop, a3 as await_block, a4 as spread_props, c as store_get, b as attr, a0 as stringify, e as escape_html, u as unsubscribe_stores } from './index-C5t0x5Ey.js';
import { b as base } from './paths-CYDIOyak.js';
import './index-CZvAxaSW.js';
import './index-_Lov4PsW.js';
import { n as navigating, p as page } from './stores-C0q5OLBG.js';
import { S as Spinner } from './Spinner-y6fbbGZF.js';
import './_commonjsHelpers-B85MJLTf.js';
import './client-CnCRRyPd.js';
import './exports-CTha0ECg.js';

function TopBar($$payload, $$props) {
  push();
  let { user, theme } = $$props;
  $$payload.out += `<div class="my-8 rounded-xl bg-white p-4 shadow-sm dark:bg-slate-700"><div class="grid grid-cols-1 gap-4 sm:mx-8 md:grid-cols-2"><div><a${attr("href", base + "/settings")} class="flex justify-center md:justify-start"><img${attr("src", `/assets/wordmark-${stringify(theme)}.png`)} class="h-8" alt="Discord Tickets"></a></div> <div><div class="mx-auto flex w-64 flex-shrink-0 items-center justify-center text-center md:float-right md:mx-0 md:justify-end md:text-left"><a${attr("href", `/auth/logout`)} class="flex items-center justify-center hover:font-medium md:justify-end" title="Logout"><img${attr("src", `https://cdn.discordapp.com/avatars/${user.id}/${user.avatar}.webp`)} class="h-8 rounded-full" alt="Discord Tickets"> <span class="ml-3">${escape_html(user.username)}</span></a> <div class="ml-4">`;
  if (theme === "dark") {
    $$payload.out += "<!--[-->";
    $$payload.out += `<i class="fa-solid fa-moon cursor-pointer p-1 text-lg transition duration-300 hover:text-blurple" title="Switch to light mode"></i>`;
  } else {
    $$payload.out += "<!--[!-->";
    $$payload.out += `<i class="fa-solid fa-sun cursor-pointer p-1 text-lg transition duration-300 hover:text-blurple" title="Switch to dark mode"></i>`;
  }
  $$payload.out += `<!--]--></div></div></div></div></div>`;
  pop();
}
const key = Symbol("modal");
function setModal(modal) {
  setContext(key, modal);
}
function StackedModalContext($$payload, $$props) {
  push();
  const { modal, children } = $$props;
  setModal(modal);
  children($$payload);
  $$payload.out += `<!---->`;
  pop();
}
class StackedModal extends EventTarget {
  static _idCounter = 0;
  _props;
  result = createDeferredPromise();
  id;
  component;
  modals;
  exitBeforeEnter = false;
  constructor(modals2, { id, component, props }) {
    super();
    this.id = id || (StackedModal._idCounter++).toString();
    this.component = component;
    this._props = props ?? {};
    this.modals = modals2;
  }
  #isOpen = once(() => {
    if (this.modals.stack.length === 0) {
      return false;
    }
    const isCurrent = this.modals.stack[this.modals.stack.length - 1].id === this.id;
    return isCurrent && !this.modals.transitioning;
  });
  get isOpen() {
    return this.#isOpen();
  }
  get index() {
    return this.modals.stack.indexOf(this);
  }
  get props() {
    return {
      ...this._props,
      id: this.id,
      index: this.index,
      isOpen: this.isOpen,
      close: this.close
    };
  }
  close = (...args) => {
    if (this.onBeforeClose?.() === false) {
      return false;
    }
    const value = args[0];
    this.dispatchEvent(new CustomEvent("close", { detail: value }));
    this.result.resolve(value);
    return true;
  };
  get promise() {
    return this.result.promise;
  }
  onBeforeClose;
}
function createDeferredPromise() {
  let resolve;
  let reject;
  const promise = new Promise((res, rej) => {
    resolve = res;
    reject = rej;
  });
  return { promise, resolve, reject };
}
class ModalStack {
  /**
   * The current stack of modals
   */
  stack = [];
  /**
   * The last action that was performed on the modals stack. This
   * can be useful for animations
   */
  action = null;
  /**
   * Whether we're currently waiting for transitions to finish
   * before opening the next modal
   */
  transitioning = false;
  /**
   * Opens a new modal
   */
  open = async (component, props, options) => {
    if (this.transitioning) {
      return;
    }
    if (options?.replace) {
      const closed = this.stack[this.stack.length - 1]?.close();
      if (!closed) {
        throw new Error("Current modal prevented closing");
      }
    }
    this.action = "push";
    const currentModal = this.stack[this.stack.length - 1];
    if (currentModal?.exitBeforeEnter) {
      this.transitioning = true;
    }
    const modal = new StackedModal(this, { id: options?.id, component, props });
    this.stack.push(modal);
    modal.addEventListener(
      "close",
      () => {
        if (modal.exitBeforeEnter) {
          this.transitioning = true;
        }
        if (this.stack.indexOf(modal) > -1) {
          this.stack.splice(this.stack.indexOf(modal), 1);
        }
        this.action = "pop";
      },
      { once: true }
    );
    return modal.promise;
  };
  /**
   * Closes the last `amount` of modals in the stack
   *
   * If closing was prevented by any modal it returns false
   */
  close = (amount = 1) => {
    if (typeof amount !== "number" || amount < 1) {
      throw new Error(`amount must be a number greater than 0. Received ${amount}`);
    }
    const closedModals = this.stack.slice(this.stack.length - amount).reverse();
    let closedAmount = 0;
    for (const modal of closedModals) {
      const closed = modal.close();
      if (!closed) {
        break;
      }
      closedAmount++;
    }
    return amount === closedAmount;
  };
  /**
   * Closes a modal by its id. Can be provided with `options.id` in modals.open(Comp, props, options)
   */
  closeById = (id) => {
    const modal = this.stack.find((modal2) => modal2.id === id);
    if (!modal) {
      return false;
    }
    return modal.close();
  };
  /**
   * Closes all modals in the stack.
   *
   * If closing was prevented by any modal, it returns false
   */
  closeAll = () => {
    return this.close(this.stack.length);
  };
}
const modals = new ModalStack();
function Modals($$payload, $$props) {
  push();
  function isLazyModal(component) {
    return typeof component.prototype === "undefined";
  }
  const { $$slots, $$events, ...props } = $$props;
  const each_array = ensure_array_like(modals.stack);
  if (modals.stack.length > 0) {
    $$payload.out += "<!--[-->";
    props.backdrop?.($$payload, modals);
    $$payload.out += `<!---->`;
  } else {
    $$payload.out += "<!--[!-->";
  }
  $$payload.out += `<!--]--> <!--[-->`;
  for (let i = 0, $$length = each_array.length; i < $$length; i++) {
    let m = each_array[i];
    StackedModalContext($$payload, {
      modal: m,
      children: ($$payload2) => {
        if (isLazyModal(m.component)) {
          $$payload2.out += "<!--[-->";
          $$payload2.out += `<!---->`;
          await_block(
            m.component(),
            () => {
              props.loading?.($$payload2, modals);
              $$payload2.out += `<!---->`;
            },
            (component) => {
              if (props.modal) {
                $$payload2.out += "<!--[-->";
                props.modal($$payload2, { component: component.default, props: m.props }, modals);
                $$payload2.out += `<!---->`;
              } else {
                $$payload2.out += "<!--[!-->";
                $$payload2.out += `<!---->`;
                component.default($$payload2, spread_props([m.props]));
                $$payload2.out += `<!---->`;
              }
              $$payload2.out += `<!--]-->`;
            }
          );
          $$payload2.out += `<!---->`;
        } else {
          $$payload2.out += "<!--[!-->";
          if (props.modal) {
            $$payload2.out += "<!--[-->";
            props.modal($$payload2, { component: m.component, props: m.props }, modals);
            $$payload2.out += `<!---->`;
          } else {
            $$payload2.out += "<!--[!-->";
            $$payload2.out += `<!---->`;
            m.component($$payload2, spread_props([m.props]));
            $$payload2.out += `<!---->`;
          }
          $$payload2.out += `<!--]-->`;
        }
        $$payload2.out += `<!--]-->`;
      }
    });
  }
  $$payload.out += `<!--]-->`;
  pop();
}
function _layout($$payload, $$props) {
  push();
  var $$store_subs;
  let { data, children } = $$props;
  const { client, user, theme } = data;
  let mounted = false;
  const links = [
    {
      icon: "fa-solid fa-book",
      name: "Documentation",
      url: "https://discordtickets.app"
    },
    {
      icon: "fab fa-discord",
      name: "Support",
      url: "https://lnk.earth/discord"
    },
    {
      icon: "fa-solid fa-comments",
      name: "Feedback",
      url: "https://lnk.earth/dsctickets-feedback"
    }
  ];
  head($$payload, ($$payload2) => {
    $$payload2.title = `<title>Discord Tickets</title>`;
    $$payload2.out += `<link rel="icon" href="/favicon.png">`;
  });
  $$payload.out += `<div class="absolute h-max min-h-screen w-full bg-gray-200 dark:bg-slate-900">`;
  {
    let backdrop = function($$payload2) {
      $$payload2.out += `<div class="backdrop"></div>`;
    }, loading = function($$payload2) {
      $$payload2.out += `<div>`;
      Spinner($$payload2);
      $$payload2.out += `<!----></div>`;
    };
    Modals($$payload, {
      backdrop,
      loading,
      $$slots: { backdrop: true, loading: true }
    });
  }
  $$payload.out += `<!----> `;
  {
    $$payload.out += "<!--[!-->";
  }
  $$payload.out += `<!--]--> <div class="m-2 sm:m-6 lg:m-12"><div class="mx-auto max-w-7xl"><div class="text-gray-800 dark:text-slate-300">`;
  if (store_get($$store_subs ??= {}, "$navigating", navigating) || !mounted) {
    $$payload.out += "<!--[-->";
    Spinner($$payload);
  } else {
    $$payload.out += "<!--[!-->";
    TopBar($$payload, { user, theme });
    $$payload.out += `<!----> `;
    children?.($$payload);
    $$payload.out += `<!----> <footer class="my-16 text-center"><div class="mb-6 p-2 text-sm"><a href="/" class="cursor-pointer text-gray-500 transition duration-300 hover:text-blurple dark:text-slate-400 dark:hover:text-blurple"><i class="fa-solid fa-arrow-left"></i> Back to the portal</a></div> `;
    if (store_get($$store_subs ??= {}, "$page", page).route.id !== "/settings") {
      $$payload.out += "<!--[-->";
      const each_array = ensure_array_like(links);
      $$payload.out += `<div class="my-4 flex justify-center gap-3"><!--[-->`;
      for (let $$index = 0, $$length = each_array.length; $$index < $$length; $$index++) {
        let link = each_array[$$index];
        $$payload.out += `<a${attr("href", link.url)} target="_blank" rel="noopener noreferrer" class="link rounded-3xl bg-gray-50/75 p-0.5 px-2 font-medium text-gray-500 shadow-sm dark:bg-slate-800/75 dark:text-slate-400"><i${attr("class", link.icon)}></i> ${escape_html(link.name)}</a>`;
      }
      $$payload.out += `<!--]--></div>`;
    } else {
      $$payload.out += "<!--[!-->";
    }
    $$payload.out += `<!--]--> <p><a href="https://discordtickets.app" target="_blank" rel="noopener noreferrer" class="cursor-pointer transition duration-300 hover:text-blurple dark:hover:text-blurple">Discord Tickets</a> by <a href="https://eartharoid.me" target="_blank" rel="noopener noreferrer" class="cursor-pointer transition duration-300 hover:text-blurple dark:hover:text-blurple">eartharoid™</a></p> <p><i class="fa-solid fa-copyright"></i> ${escape_html((/* @__PURE__ */ new Date()).getFullYear())} <a href="https://eartharoid.me" target="_blank" class="cursor-pointer transition duration-300 hover:text-blurple dark:hover:text-blurple">Isaac Saunders</a></p> <p class="my-4 text-xs">Discord Tickets is not an official Discord product. <br> It is not affiliated with nor endorsed by Discord Inc.</p></footer>`;
  }
  $$payload.out += `<!--]--></div></div></div></div>`;
  if ($$store_subs) unsubscribe_stores($$store_subs);
  pop();
}

export { _layout as default };
