// leekslazylogger@6.0.0/dist/transports/file/defaults.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const dtf_1 = __importDefault(require("@eartharoid/dtf"));
const dtf = new dtf_1.default('en-GB');
const { version } = require('../../../package.json');
const defaults = {
    clean_directory: 7,
    directory: './logs',
    file: 'YYYY-MM-DD.log',
    format: function (log) {
        const timestamp = typeof this.timestamp === 'function' ? this.timestamp(log.timestamp) : dtf.fill(this.timestamp, log.timestamp);
        return `[${timestamp}] [${log.level.name.toUpperCase()}] ${log.namespace ? `(${log.namespace.toUpperCase()}) ` : ''}${log.content}`;
    },
    header: function () {
        const datetime = dtf.fill('DDDD, DD MMMM YYYY at HH:mm AMPM');
        const timezone = Intl.DateTimeFormat().resolvedOptions().timeZone;
        return `\r\n\t❯ ${this.name}\r\n\t❯ Powered by leekslazylogger v${version}\r\n\t❯ Log level "${this.level}"\r\n\t❯ Node.js ${process.version} on ${process.platform}\r\n\t❯ ${datetime} (${timezone}) -->\r\n`;
    },
    level: 'info',
    name: 'A leekslazylogger project',
    new_file: 'day',
    timestamp: 'DD/MM/YY HH:mm:ss',
};
exports.default = defaults;
