// leekslazylogger@6.0.0/dist/transports/console/defaults.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const leeks_js_1 = require("leeks.js");
const dtf_1 = __importDefault(require("@eartharoid/dtf"));
const dtf = new dtf_1.default('en-GB');
const defaults = {
    colours: {
        critical: '&!4&0',
        debug: '&1',
        error: '&4',
        info: '&3',
        notice: '&!6&0',
        success: '&2',
        warn: '&6',
    },
    format: function (log) {
        const timestamp = typeof this.timestamp === 'function' ? this.timestamp(log.timestamp) : dtf.fill(this.timestamp, log.timestamp);
        const colour = this.colours[log.level.name] ?? '';
        return (0, leeks_js_1.short)(`${colour}[${timestamp}] [${log.level.name.toUpperCase()}] ${log.namespace ? `(${log.namespace.toUpperCase()}) ` : ''}${log.content}`);
    },
    level: 'info',
    timestamp: 'DD/MM/YY HH:mm:ss',
};
exports.default = defaults;
