// @eartharoid/dbf@0.4.2/dist/commands/SlashCommand.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const Command_1 = __importDefault(require("./Command"));
const discord_js_1 = require("discord.js");
class SlashCommand extends Command_1.default {
    constructor(client, options) {
        super(client, Object.assign({ type: discord_js_1.ApplicationCommandType.ChatInput }, options));
        this.options = options.options;
    }
    toJSON() {
        return {
            defaultMemberPermissions: this.defaultMemberPermissions,
            description: this.description,
            descriptionLocalizations: this.descriptionLocalizations,
            dmPermission: this.dmPermission,
            name: this.name,
            nameLocalizations: this.nameLocalizations,
            options: this.options,
            type: this.type,
        };
    }
}
exports.default = SlashCommand;
