// @discord-tickets/settings@2.4.0/build/server/manifest.js
const manifest = (() => {
function __memo(fn) {
	let value;
	return () => value ??= (value = fn());
}

return {
	appDir: "_app",
	appPath: "_app",
	assets: new Set(["assets/topgg-dark.png","assets/topgg-light.png","assets/undraw_reviews.svg","assets/wordmark-dark.png","assets/wordmark-light.png","favicon.png"]),
	mimeTypes: {".png":"image/png",".svg":"image/svg+xml"},
	_: {
		client: {"start":"_app/immutable/entry/start.iHWq1yft.js","app":"_app/immutable/entry/app.zpSLmJsW.js","imports":["_app/immutable/entry/start.iHWq1yft.js","_app/immutable/chunks/entry.DD3pBHu9.js","_app/immutable/chunks/index-client.BK-sKD8p.js","_app/immutable/chunks/runtime.D2_qJMqH.js","_app/immutable/chunks/index.BFqC5wTN.js","_app/immutable/chunks/utils.ZaDWqmk0.js","_app/immutable/chunks/paths.C8eWa2kM.js","_app/immutable/chunks/control.CYgJF_JY.js","_app/immutable/entry/app.zpSLmJsW.js","_app/immutable/chunks/preload-helper.C1FmrZbK.js","_app/immutable/chunks/legacy-client.bwmqDaLy.js","_app/immutable/chunks/runtime.D2_qJMqH.js","_app/immutable/chunks/index.BFqC5wTN.js","_app/immutable/chunks/render.DYyZBxIN.js","_app/immutable/chunks/utils.Dghph3y9.js","_app/immutable/chunks/svelte-head.BxbvRlmQ.js","_app/immutable/chunks/template.-1lJmaN6.js","_app/immutable/chunks/disclose-version.Bg9kRutz.js","_app/immutable/chunks/if.DQDEx0g4.js","_app/immutable/chunks/svelte-component.BdD4hKdB.js","_app/immutable/chunks/proxy.CdfeBWYI.js","_app/immutable/chunks/props.DioveIhQ.js","_app/immutable/chunks/store.B-TJ19st.js","_app/immutable/chunks/utils.ZaDWqmk0.js","_app/immutable/chunks/index-client.BK-sKD8p.js"],"stylesheets":[],"fonts":[],"uses_env_dynamic_public":false},
		nodes: [
			__memo(() => import('./chunks/0-Liwzr___.js')),
			__memo(() => import('./chunks/1-njWT8Gay.js')),
			__memo(() => import('./chunks/2--Mo52QOm.js')),
			__memo(() => import('./chunks/3-DLvoTGM4.js')),
			__memo(() => import('./chunks/4-BYfuWGm4.js')),
			__memo(() => import('./chunks/5-C_jtX0gJ.js')),
			__memo(() => import('./chunks/6-CCFYEjNU.js')),
			__memo(() => import('./chunks/7-C3FSOYGW.js')),
			__memo(() => import('./chunks/8-BPNxFVix.js')),
			__memo(() => import('./chunks/9-D3b6RL3m.js')),
			__memo(() => import('./chunks/10-NTCqru06.js')),
			__memo(() => import('./chunks/11-C02GIQv0.js')),
			__memo(() => import('./chunks/12-Cxi73DIP.js')),
			__memo(() => import('./chunks/13-CKDMG9i9.js')),
			__memo(() => import('./chunks/14-CkgeptZQ.js')),
			__memo(() => import('./chunks/15-D7p7QfuU.js')),
			__memo(() => import('./chunks/16-Cl5OWL0f.js')),
			__memo(() => import('./chunks/17-BARUUtYf.js')),
			__memo(() => import('./chunks/18-nvEMqBvZ.js')),
			__memo(() => import('./chunks/19-CWfphEZ5.js')),
			__memo(() => import('./chunks/20-BeCWOHz4.js')),
			__memo(() => import('./chunks/21-C4I6cN91.js')),
			__memo(() => import('./chunks/22-CyKFTM-k.js')),
			__memo(() => import('./chunks/23-Db3XzeIj.js'))
		],
		routes: [
			{
				id: "/(default)",
				pattern: /^\/?$/,
				params: [],
				page: { layouts: [0,2,], errors: [1,3,], leaf: 8 },
				endpoint: null
			},
			{
				id: "/(default)/login",
				pattern: /^\/login\/?$/,
				params: [],
				page: { layouts: [0,2,], errors: [1,3,], leaf: 9 },
				endpoint: null
			},
			{
				id: "/settings",
				pattern: /^\/settings\/?$/,
				params: [],
				page: { layouts: [0,5,], errors: [1,6,], leaf: 15 },
				endpoint: null
			},
			{
				id: "/settings/[guild]",
				pattern: /^\/settings\/([^/]+?)\/?$/,
				params: [{"name":"guild","optional":false,"rest":false,"chained":false}],
				page: { layouts: [0,5,], errors: [1,6,], leaf: 16 },
				endpoint: null
			},
			{
				id: "/settings/[guild]/categories",
				pattern: /^\/settings\/([^/]+?)\/categories\/?$/,
				params: [{"name":"guild","optional":false,"rest":false,"chained":false}],
				page: { layouts: [0,5,7,], errors: [1,6,,], leaf: 17 },
				endpoint: null
			},
			{
				id: "/settings/[guild]/categories/[category]",
				pattern: /^\/settings\/([^/]+?)\/categories\/([^/]+?)\/?$/,
				params: [{"name":"guild","optional":false,"rest":false,"chained":false},{"name":"category","optional":false,"rest":false,"chained":false}],
				page: { layouts: [0,5,7,], errors: [1,6,,], leaf: 18 },
				endpoint: null
			},
			{
				id: "/settings/[guild]/feedback",
				pattern: /^\/settings\/([^/]+?)\/feedback\/?$/,
				params: [{"name":"guild","optional":false,"rest":false,"chained":false}],
				page: { layouts: [0,5,7,], errors: [1,6,,], leaf: 19 },
				endpoint: null
			},
			{
				id: "/settings/[guild]/general",
				pattern: /^\/settings\/([^/]+?)\/general\/?$/,
				params: [{"name":"guild","optional":false,"rest":false,"chained":false}],
				page: { layouts: [0,5,7,], errors: [1,6,,], leaf: 20 },
				endpoint: null
			},
			{
				id: "/settings/[guild]/panels",
				pattern: /^\/settings\/([^/]+?)\/panels\/?$/,
				params: [{"name":"guild","optional":false,"rest":false,"chained":false}],
				page: { layouts: [0,5,7,], errors: [1,6,,], leaf: 21 },
				endpoint: null
			},
			{
				id: "/settings/[guild]/reset",
				pattern: /^\/settings\/([^/]+?)\/reset\/?$/,
				params: [{"name":"guild","optional":false,"rest":false,"chained":false}],
				page: { layouts: [0,5,7,], errors: [1,6,,], leaf: 22 },
				endpoint: null
			},
			{
				id: "/settings/[guild]/tags",
				pattern: /^\/settings\/([^/]+?)\/tags\/?$/,
				params: [{"name":"guild","optional":false,"rest":false,"chained":false}],
				page: { layouts: [0,5,7,], errors: [1,6,,], leaf: 23 },
				endpoint: null
			},
			{
				id: "/(default)/view/[ticket]",
				pattern: /^\/view\/([^/]+?)\/?$/,
				params: [{"name":"ticket","optional":false,"rest":false,"chained":false}],
				page: { layouts: [0,2,], errors: [1,3,], leaf: 10 },
				endpoint: null
			},
			{
				id: "/(default)/[guild]",
				pattern: /^\/([^/]+?)\/?$/,
				params: [{"name":"guild","optional":false,"rest":false,"chained":false}],
				page: { layouts: [0,2,4,], errors: [1,3,,], leaf: 11 },
				endpoint: null
			},
			{
				id: "/(default)/[guild]/feedback",
				pattern: /^\/([^/]+?)\/feedback\/?$/,
				params: [{"name":"guild","optional":false,"rest":false,"chained":false}],
				page: { layouts: [0,2,4,], errors: [1,3,,], leaf: 12 },
				endpoint: null
			},
			{
				id: "/(default)/[guild]/staff",
				pattern: /^\/([^/]+?)\/staff\/?$/,
				params: [{"name":"guild","optional":false,"rest":false,"chained":false}],
				page: { layouts: [0,2,4,], errors: [1,3,,], leaf: 13 },
				endpoint: null
			},
			{
				id: "/(default)/[guild]/tickets",
				pattern: /^\/([^/]+?)\/tickets\/?$/,
				params: [{"name":"guild","optional":false,"rest":false,"chained":false}],
				page: { layouts: [0,2,4,], errors: [1,3,,], leaf: 14 },
				endpoint: null
			}
		],
		matchers: async () => {
			
			return {  };
		},
		server_assets: {}
	}
}
})();

const prerendered = new Set([]);

const base = "";

export { base, manifest, prerendered };
